src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -35,23 +46,42 @@ export function useRefetchQuoteCallback() {
         : undefined
 
       const [fee, price] = await Promise.all([feePromise, pricePromise])
-      if ((fee || !fetchFee) && price) {
+      // use typecheck on returned fee/price response
+      // can be a valid FeeInformation/Price object
+      // or an UnsupportedToken error response
+      if ((_validResponse(fee) || !fetchFee) && _validResponse(price)) {
         // Update quote
+        // fee.data can still be undefined
+        // price.data CANNOT be undefined (L22)
+        // TODO: check this
         updateQuote({
           sellToken,
           buyToken,
           amount,
-          price,
+          // TS hates the typecheck on L52, price.data is not undefined
+          price: price?.data as PriceInformation,
           chainId,
           lastCheck: Date.now(),
           // Fee is only updated when fetchFee=true
-          fee
+          fee: fee?.data
+        })
+      } else if ((price && !_validResponse(price)) || (fee && !_validResponse(fee))) {
+        const unsupportedToken = (price?.error || fee?.error) as UnsupportedToken
+        // unsupported token error, mark token as such
+        // TODO: will likely change with server response including a data.address
+        // field for UnsupportedToken error responses
+        const address = unsupportedToken.description.split(' ')[2]
+        console.debug('[UNSUPPORTED TOKEN!]::', address)
+        if (unsupportedTokensMap?.[address]) return
```

Comment:
this if return is a bit weird in the middle. Also dangerous practice. I thought it was a bad indentation and it was protecting addUnsupportedToken

u did that to avoid negating the IF condition?